students/LincolnZ/lesson01/assignment/tests/test_unit.py

```diff
@@ -0,0 +1,126 @@
+"""unit test"""
+from unittest import TestCase
+from unittest.mock import patch
+from unittest.mock import MagicMock
+from inventory_management.inventory_class import Inventory
+from inventory_management.furniture_class import Furniture
+from inventory_management.electric_appliance_class import ElectricAppliances
+from inventory_management import main
+from inventory_management import market_prices
+
+
+class inventory_test(TestCase):
+    """unit test for inventory and its sub classes"""
+
+    def setUp(self):
+        """set up instance of each sub class"""
+        self.furniture = Furniture(100, "test chair",
+                                   20, (1.5, "day"),
+                                   "wood", "Large")
+
+        self.appliance = ElectricAppliances(200, "test lamp",
+                                            30, (2.5, "day"),
+                                            "basic", "110V")
+        self.appliance_expected_output = {
+                                            "product_code": 200,
+                                            "description": "test lamp",
+                                            "market_price": 30,
+                                            "rental_price": (2.5, "day"),
+                                            "brand": "basic",
+                                            "voltage": "110V",
+                                            }
+        self.furniture_expected_output = {
+                                            "product_code": 100,
+                                            "description": "test chair",
+                                            "market_price": 20,
+                                            "rental_price": (1.5, "day"),
+                                            "material": "wood",
+                                            "size": "Large",
+                                            }
+
+    def test_electric_appliances(self):
+
+        self.assertDictEqual(self.appliance.return_as_dictionary(), self.appliance_expected_output)
+
+    def test_furniture(self):
+
+        self.assertDictEqual(self.furniture.return_as_dictionary(), self.furniture_expected_output)
+
+
+class test_market_price(TestCase):
+
+    def test_get_latest_price_mock(self):
+        """a mock test"""
+        market_prices.get_latest_price = MagicMock(return_value=3)
+        market_prices.get_latest_price()
+        market_prices.get_latest_price.assert_called()
```

Comment:
You can probably get rid of this one since the one below asserts that it's been called and only once